### PR TITLE
fix: Show loading progress while parsing local JSON assets

### LINKED ISSUE #1703: Show progress while static JSON assets load
**Actual Behaviour**

There is no progress while static assets are loaded, along with the clearing of versions on loading from API

**Expected Behaviour**

Show progress while parsing of JSON assets and clearing of API versions when loading from server

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #1703 

Changes: 
- Add verbose error handling during loading of resources
- Add correct count to event while displaying progress bar
- Add progress bar when loading static assets
- Show currently downloading and previously downloaded data
- Notify UI to refresh once the downloading is complete

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @shreyasnbhat @mahikaw Please review

**Comment by Shailesh351** [comment]
@iamareebjamal All working fine. but getting one problem for root url empty case.

After uninstalling app if i again run app directly from android studio then only once app is getting stuck i mean download progress dialog. Then if i close app and open app again then all the data from assets are not loaded. only that much data is loaded which was saved before dialog stuck.

**Comment by iamareebjamal** [comment]
OK. I'm getting it, but can you explain the part about the dialog being stuck. I mean, is it stuck at 0, or something else, and can you see the logs and post here. Till the app gets stuck

**Comment by Shailesh351** [comment]
Sometimes it stops at 0 sometimes some other values.

`06-13 22:06:35.107 12138-12307/org.fossasia.openevent I/OpenGLRenderer: Initialized EGL, version 1.4
06-13 22:06:35.107 12138-12307/org.fossasia.openevent D/OpenGLRenderer: Swap behavior 1
06-13 22:06:35.269 12138-12138/org.fossasia.openevent D/DownloadCounter: 6 counter
06-13 22:06:35.270 12138-12138/org.fossasia.openevent D/DownloadCompleteHandler: Progress : 0.000000 Downloading Event Info…
06-13 22:06:35.358 12138-12145/org.fossasia.openevent I/art: Do partial code cache collection, code=17KB, data=30KB
06-13 22:06:35.359 12138-12145/org.fossasia.openevent I/art: After code cache collection, code=17KB, data=30KB
06-13 22:06:35.359 12138-12145/org.fossasia.openevent I/art: Increasing code cache capacity to 128KB
06-13 22:06:35.612 12138-12138/org.fossasia.openevent I/BaseActivity: Activity onResume
`

**Comment by iamareebjamal** [comment]
This happens if the root url is empty?

**Comment by Shailesh351** [comment]
Yes, In that case

**Comment by iamareebjamal** [comment]
In my case, if there is no root URL, the JSON is automatically loaded as you coded it to @Shailesh351 mm

**Comment by iamareebjamal** [comment]
Weren't you doing a clean install after deleting the root URL?

**Comment by iamareebjamal** [comment]
@arp95 @mahikaw @rohanagarwal94 This PR is open for 2 days. I know you guys are super busy, but still, we need to reduce the time between PR merges :smile:

**Comment by arp95** [comment]
@iamareebjamal really sorry for that. i am quite busy with samples nowadays.
anyways having a look.

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 1]
```diff
@@ -372,11 +377,17 @@ private void setNavHeader(Event event) {
     }
 
     private void syncComplete() {
-        // Event successfully loaded, set data downloaded to true
-        sharedPreferences.edit().putBoolean(ConstantStrings.IS_DOWNLOAD_DONE, true).apply();
+        String successMessage = "Data loaded from JSON";
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: please use values from strings.xml

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 2]
```diff
@@ -748,6 +774,11 @@ public void downloadFromAssets() {
             //TODO: Add and Take counter value from to config.json
             sharedPreferences.edit().putBoolean(ConstantStrings.DATABASE_RECORDS_EXIST, true).apply();
 
+            startDownloadListener();
+            Timber.d("JSON parsing started");
+
+            OpenEventApp.postEventOnUIThread(new CounterEvent(6)); // Bump if increased
```
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: the number 6 is random?
i mean how did you decide what to use?

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 1 above)
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: This is for logging, not to show users a message

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 2 above)
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: 1. Event
2. Sessions
3. Tracks
4. Speakers
5. Sponsors
6. Microlocations

**Comment by Shailesh351** [comment]
I'm not facing this issue now. May be some issue with my device before.

**Comment by Shailesh351** [comment]
This is how progress is being showed 

D/DownloadCompleteHandler: Progress : 0.000000 Downloading Event Info…
D/DownloadCompleteHandler: Progress : 16.666668 Downloading Event Info…
D/DownloadCompleteHandler: Progress : 33.333336 Downloading Sponsors…
D/DownloadCompleteHandler: Progress : 50.000000 Downloading Locations…
D/DownloadCompleteHandler: Progress : 66.666672 Downloading Tracks…
D/DownloadCompleteHandler: Progress : 83.333328 Downloading Speakers…
D/DownloadCompleteHandler: Progress : 100.000000 Downloading Sessions…

Shouldn't it be like 

D/DownloadCompleteHandler: Progress : 0.000000 Downloading Event Info…
D/DownloadCompleteHandler: Progress : 16.666668 Downloading Sponsors…
D/DownloadCompleteHandler: Progress : 33.333336 Downloading Locations…
D/DownloadCompleteHandler: Progress : 50.000000 Downloading Tracks…
D/DownloadCompleteHandler: Progress : 66.666672 Downloading Speakers…
D/DownloadCompleteHandler: Progress : 83.333328 Downloading Sessions…
D/DownloadCompleteHandler: Progress : 100.000000 Download complete.

?

**Comment by iamareebjamal** [comment]
True. Will look into it

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 2 above)
Comment (arp95) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: what about session_types json file?

**Comment by iamareebjamal** [comment]
@Shailesh351 Please check now

**Comment by iamareebjamal** [comment]
@arp95 @mahikaw @rohanagarwal94 @mejariamol Please review

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 2 above)
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: We neither load it from the internet, nor from JSON

Please create an issue for this. To be solved later

**Comment by iamareebjamal** [comment]
@the-dagger Please review and merge

**Pull request merged**